Updates to cookie text (#169)
tlswg/dtls13-spec: draft-ietf-tls-dtls13.md

the client sends its ClientHello message to the server, the server
MAY respond with a HelloRetryRequest message. The HelloRetryRequest message,
as well as the cookie extension, is defined in TLS 1.3.
The HelloRetryRequest message contains a stateless cookie (see

> {{RFC8446}}; Section 4.2.2.).

Reviewer: ```suggestion
{{RFC8446}}; Section 4.2.2).
```

the client sends its ClientHello message to the server, the server
MAY respond with a HelloRetryRequest message. The HelloRetryRequest message,
as well as the cookie extension, is defined in TLS 1.3.
The HelloRetryRequest message contains a stateless cookie (see

> {{RFC8446}}; Section 4.2.2).

Reviewer 2: ```suggestion
{{!TLS13}}; Section 4.2.2).
```

that the exchange is performed, however.  In addition, the server MAY
choose not to do a cookie exchange when a session is resumed or, more
generically, when the DTLS handshake uses a PSK-based key exchange.
Servers which process 0-RTT requests and send 0.5-RTT responses

> without a cookie exchange should

Reviewer: SHOULD?

Isn't the alternative here to acknowledge the possibility that 0.5-RTT response are sent to an address that has not demonstrated a willingness to communicate.

Author: Do you have proposed text.

> store the client's IP in the cookie and limit the size of their

Reviewer: IMPORTANT: you just said that no cookie exchange is involved.  I think that you mean session ticket.

```suggestion
store the client's IP in the session ticket and limit the size of their
```

Author: yeah, clearly it's getting too late in the day for me to edit.

> Servers which process 0-RTT requests and send 0.5-RTT responses
> without a cookie exchange should
> store the client's IP in the session ticket and limit the size of their
> responses in order to avoid amplification attacks.

Reviewer: ```suggestion
Servers which process 0-RTT requests and send 0.5-RTT responses
without a cookie exchange risk being used in an amplification attack
if the size of outgoing greatly exceeds the size of those that are received.
A server SHOULD limit the amount of data it sends toward a client address
before it verifies that the client is able to receive data at that address.
A client address is valid after a cookie exchange or handshake completion.
A server MAY apply a higher limit based on heuristics, such as if the client
uses the same IP address as a recent connection.
```

Reviewer 2: This suggested text reads fine to me. Do we want to also include @ekr's suggestion that the ticket store the client's address, if only as a way of checking "if the client uses the same IP address as a recent connection"?

that the exchange is performed, however.  In addition, the server MAY
choose not to do a cookie exchange when a session is resumed or, more
generically, when the DTLS handshake uses a PSK-based key exchange.
Servers which process 0-RTT requests and send 0.5-RTT responses
without a cookie exchange risk being used in an amplification attack

> if the size of outgoing greatly exceeds the size of those that are received.

Reviewer 3: ```suggestion
if the size of outgoing messages greatly exceeds the size of those that are received.
```

A server SHOULD limit the amount of data it sends toward a client address
before it verifies that the client is able to receive data at that address.
A client address is valid after a cookie exchange or handshake completion.
A server MAY apply a higher limit based on heuristics, such as if the client

> uses the same IP address as a recent connection.

Reviewer 4: I wonder whether it makes sense to put out that this is not the regular amplification attack because the 0-RTT message is authenticated. While there is a risk of replay attack here, this assumes that 
(a) the attacker has been able to observe the initial 0-RTT request, or 
(b) the attacker is the endpoint transmitting the 0-RTT request itself. 

In case (b), if the endpoint was authenticated in the initial handshake then it is fairly easy to determine the source of the attack. In a classical amplification attack the adversary would probably also modify the source IP address to conceal its location, which does not help much here.  

Am I seeing this wrong?

Author: No I think you're right, but I think the text covers it reasonably well.
